packages/backend/src/request/requestContextProvider.ts

```diff
@@ -1,20 +1,24 @@
 import * as Express from 'express';
-import { AppSingleton, ServiceContext, App } from '@h4bff/core';
+import { AppSingleton, ServiceContext, App, ServiceContextEvents } from '@h4bff/core';
 import { RequestInfo } from './';
 
 /**
- * Ties each request / response pair to a specific service context.
+ * Ties each provided request / response pair to a specific service context.
  */
 export class RequestContextProvider extends AppSingleton {
-  router = Express();
-
   private contexts = new WeakMap<Express.Request, ServiceContext>();
 
   constructor(app: App) {
     super(app);
-    this.router.use(this.contextualWrapper);
+    app.getSingleton(ServiceContextEvents).onContextDisposed((sc, _error) => {
+      return this.onDispose(sc);
+    });
   }
 
+  /**
+   * Returns the context bound to the provided request object.
+   * If it doesn't exists, it creates a new context and return.
+   */
   public getContext(req: Express.Request, res: Express.Response) {
```

Comment:
only safe withRequestContext API should be exposed, if possible.

Reply:
Since `getContext`is used in the `RPCServiceRegistry` it is not possible to make it `private` at the moment. I would tackle this in a separate PR.